### PR TITLE
refactor: convert bp.settings into multiple tables

### PR DESCRIPTION
**Pull request by klesh**
### Summary
refactor: convert bp.settings into multiple tables

### Does this close any open issues?
  Closes #5655, #5360

### Screenshots

Snipaste_2023-08-31_14-43-56
trigger-pipeline

### CODE REVIEW

File: backend/core/models/migrationscripts/20230829_normalize_bp_settings.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		new(blueprint20230829),
+		new(archived.BlueprintConnection),
+		new(archived.BlueprintScope),
+	)
+	if err != nil {
+		return err
+	}
+	encKey := basicRes.GetConfig(plugin.EncodeKeyEnvStr)
+	if encKey == "" {
+		return errors.BadInput.New("jira v0.11 invalid encKey")
```
Comment (abeizn) [inline backend/core/models/migrationscripts/20230829_normalize_bp_settings.go:73]: jira v0.11 ?

File: backend/core/services/pipeline_service.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// PipelineQuery is a query for GetPipelines
+// type PipelineQuery struct {
+// 	Pagination
+// Label string `form:"label"`
+// }
+
+type PipelineService interface {
+	Create(bp *models.NewPipeline) (*models.Pipeline, errors.Error)
+	// Query(query *PipelineQuery) (*Paginated[PipelineQuery], errors.Error)
+	// Get(id uint64) (*models.Pipeline, errors.Error)
```
Comment (abeizn) [inline backend/core/services/pipeline_service.go:34]: Are these interfaces unnecessary?

File: backend/core/models/migrationscripts/20230829_normalize_bp_settings.go (hunk 1 above)
Comment (klesh) [reply backend/core/models/migrationscripts/20230829_normalize_bp_settings.go:73]: good catch

File: backend/core/services/pipeline_service.go (hunk 2 above)
Comment (klesh) [reply backend/core/services/pipeline_service.go]: good catch

**Review by abeizn — APPROVED** [review]

**Pull request merged**